Tasks/Maven/maventask.ts

```diff
@@ -227,14 +227,43 @@ function applyPmdGoals(mvnRun: trm.ToolRunner):void {
 }
 
 // Returns true if any one code analysis tool was enabled.
-function isCodeAnalysisReportRequired():boolean {
+function isCodeAnalysisUploadRequired():boolean {
     return tl.getInput('pmdAnalysisEnabled', true) == 'true';
 }
 
+// Take a result object and attempt to upload the XML and HTML reports as artifacts. Throws if fail.
+function uploadBuildArtifactsFromResult(analysisResult:ar.AnalysisResult):void {
+    // make sure the staging folder exists and is empty
+    tl.rmRF(stagingDir);
+    tl.mkdirP(stagingDir);
+
+    var filesToUpload:string[] = [];
+    if (analysisResult.xmlFilePath) {
+        filesToUpload.push(analysisResult.xmlFilePath)
+    }
+    if (analysisResult.htmlFilePath) {
+        filesToUpload.push(analysisResult.htmlFilePath);
+    }
+
+    filesToUpload.forEach(fileToUpload => {
+        // Need copy files to the staging folder and upload the entire folder for compatibility with the Xplat agent
+        var stagingFilePath = path.join(stagingDir, path.basename(fileToUpload));
+        tl.debug('Staging ' + fileToUpload + ' to ' + stagingFilePath);
+        tl.cp('-f', fileToUpload, stagingFilePath);
+
+
+        if (!tl.exist(stagingFilePath)) {
+            console.warn('Could not find file to upload: ' + stagingFilePath)
```

Comment:
Not sure if this is the right place to throw this kind of warning, i.e. it is too generic. If you can catch it earlier, when you create the analysis result structure, you can give a better warning message.